### PR TITLE
idat: protect against trying to output negative size

### PR DESCRIPTION
**Pull request by bradh**
This one showed up in box fuzzing.

I wasn't sure about creating a new unit test - can drop that part if you would prefer.

### CODE REVIEW

File: libheif/box.cc [hunk 1]
```diff
@@ -2854,7 +2854,9 @@ std::string Box_idat::dump(Indent& indent) const
   std::ostringstream sstr;
   sstr << Box::dump(indent);
 
-  sstr << indent << "number of data bytes: " << get_box_size() - get_header_size() << "\n";
+  if (get_box_size() > get_header_size()) {
+    sstr << indent << "number of data bytes: " << get_box_size() - get_header_size() << "\n";
+  }
```
Comment (farindk) [inline libheif/box.cc:2859]: Could we add something like
````
else {
  sstr << indent << "number of data bytes is invalid\n";
}
````

Comment (farindk) [reply libheif/box.cc:2859]: Maybe `>=` ? Size 0 might be a valid edge case.

Comment (bradh) [reply libheif/box.cc:2859]: > Maybe `>=` ? Size 0 might be a valid edge case.

I checked 14496-12:2022 Section 8.11.11 and didn't see any prohibition, so assume that 0 is valid. Have updated the code.

Comment (bradh) [reply libheif/box.cc:2859]: Have updated the PR to include these.

**Comment by farindk** [comment]
Thanks

**Pull request merged**